static/js/hoc/withChannelTracker.js

```diff
@@ -0,0 +1,37 @@
+// @flow
+import React from "react"
+import ReactGA from "react-ga"
+
+export const withChannelTracker = (
+  WrappedComponent: Class<React.Component<*, *>>
+) => {
+  class WithChannelTracker extends React.Component<*, *> {
+    isTracked = false
+
+    loadGA() {
+      const { channel, location } = this.props
+      if (channel && channel.ga_tracking_id && !this.isTracked) {
+        const trackerName = channel.ga_tracking_id.replace(/-/g, "_")
+        ReactGA.ga("create", channel.ga_tracking_id, "auto", {
+          name: trackerName
+        })
+        ReactGA.ga(`${trackerName}.send`, "pageview", location.pathname)
+        this.isTracked = true
```

Comment:
Hmmm, gotta be careful with this. I *think* this breaks the tracking if someone switches from channel A to channel B because the component never reinitializes. We ran into similar issue with `ChannelPage` and solved it with this: https://github.com/mitodl/open-discussions/blob/master/static/js/containers/ChannelPage.js#L73

Reply:
Confirmed that this definitely happens.